From 981b20634a7a915ab53b26df4f3c0b22f5b4caa4 Mon Sep 17 00:00:00 2001
From: Xenia Ragiadakou <xenia.ragiadakou@amd.com>
Date: Fri, 13 Oct 2023 22:30:28 +0300
Subject: [PATCH 089/106] x86/hvm: do not invalidate mapcache when removing
 non-ram mappings

Access to foreign/granted pages is never emulated by ioreq. Hence, do not
invalidate foreign/granting domain's mapcache when removing such mappings.
Request mapcache invalidation only for ram-type p2m mappings.

Signed-off-by: Xenia Ragiadakou <xenia.ragiadakou@amd.com>
Change-Id: Id2fed8ffb0fe35e00305888ed32f9b1805d9182c
---
 xen/arch/x86/mm/p2m.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xen/arch/x86/mm/p2m.c b/xen/arch/x86/mm/p2m.c
index e9f2fde821..8fa10d9d8a 100644
--- a/xen/arch/x86/mm/p2m.c
+++ b/xen/arch/x86/mm/p2m.c
@@ -535,8 +535,8 @@ p2m_remove_entry(struct p2m_domain *p2m, gfn_t gfn, mfn_t mfn,
         }
     }
 
-    //TODO fix for dma
-    ioreq_request_mapcache_invalidate(p2m->domain);
+    if ( p2m_is_ram(t) )
+        ioreq_request_mapcache_invalidate(p2m->domain);
 
     rc = p2m_set_entry(p2m, gfn, INVALID_MFN, page_order, p2m_invalid,
                        p2m->default_access);
-- 
2.25.1

